ntbs-service/Services/TestResultsRepository.cs

```diff
@@ -26,9 +29,17 @@ public async Task AddTestResultAsync(ManualTestResult testResult)
             await UpdateDatabaseAsync();
         }
 
-        public async Task UpdateTestResultAsync(ManualTestResult testResult)
+        public async Task UpdateTestResultAsync(Notification Notification, ManualTestResult testResult)
         {
-            _context.Attach(testResult);
```

Comment:
Why did you change this? Did it not update properly, or was it an auditing thing? I feel like functionally both should be equivalent

Reply:
Because I already had the context-attached model, and I was trying to attach another instance with the same id. The way it does it now follows what we do for notification - we pass in the context-attached model, and a model with the new data we want.

I've not spotted it before because my add/update logic was backwards 🤦‍♂️

Reply:
I see, makes sense!